### PR TITLE
Update mongo lock to new driver

### PR DESCRIPTION
**Pull request by fredtsun**
# JIRA
https://clever.atlassian.net/browse/DING-1795

# Overview
this pr upgrades the deprecated mongo driver, mgo, to use mongo-go-driver

# Testing
Unit tested
#

### CODE REVIEW

File: rw_mutex.go [hunk 1]
```diff
@@ -29,8 +31,30 @@ type mongoLock struct {
...
+	"$or": []bson.M{
+		{
+			"readers": nil,
+		},
+		{
+			"readers": []string{},
+		},
+	},
+}
+
```
Comment (SpiffyUrsa) [inline rw_mutex.go:55]: https://github.com/Clever/mongo-lock-go/pull/22/files#diff-8044f62efb4b1121122df89a0dee1859d30c8bbbcba9d7c83958b23e48eea8d8R34-R55

For this part, is this just typing what the different lock types can be?

File: rw_mutex.go [hunk 2]
```diff
@@ -156,61 +179,54 @@ func (m *RWMutex) tryToGetReadLock(lock *mongoLock) error {
...
-			err := m.collection.
-				Find(bson.M{"lockID": m.lockID}).
-				One(&lock)
-			return &lock, err
-		} else if err != nil {
-			return nil, err
+
+	res := m.collection.FindOneAndUpdate(context.TODO(),
+		bson.M{"lockID": m.lockID}, bson.M{"$set": bson.M{"lockID": m.lockID}},
+		options.FindOneAndUpdate().SetReturnDocument(options.After).SetUpsert(true),
```
Comment (SpiffyUrsa) [inline rw_mutex.go:224]: Can you maybe add a comment as to what this is doing?

File: rw_mutex.go (hunk 1 above)
Comment (fredtsun) [reply rw_mutex.go:55]: oh it's just what an empty read/write lock query would look like. i.e. if we needed to query mongo to see if a read lock is available, the readers portion could be nil or and empty array, so we need to check both cases.

Comment (SpiffyUrsa) [reply rw_mutex.go:55]: Gotcha thanks!

File: rw_mutex.go (hunk 2 above)
Comment (SpiffyUrsa) [reply rw_mutex.go:224]: Thanks for doing that!

**Review by SpiffyUrsa — APPROVED** [review]
LGTM!

**Pull request merged**